src/main/frontend/handler/editor.cljs

```diff
@@ -2734,14 +2734,14 @@
           (js/document.execCommand "copy"))
         (delete-and-update input selected-start selected-end))
 
-      (and (zero? current-pos)
-           ;; not the top block in a block page
-           (not (and page
-                     (util/uuid-string? page)
-                     (= (medley/uuid page) block-id))))
+      (zero? current-pos)
       (do
         (util/stop e)
-        (delete-block! repo false))
+        (when-not (and page
```

Comment:
This condition fails on journal pages. should also check `:block/journal?`

Reply:
```clojure
(do
  (util/stop e)
  (when-not (and (or page (:block/journal? block))
                 (= (:block/page block) (:block/left block)))
    (delete-block! repo false)))
```

Reply:
Got it! I'll fix it soon.

Reply:
> ```clojure
> (do
>   (util/stop e)
>   (when-not (and (or page (:block/journal? block))
>                  (= (:block/page block) (:block/left block)))
>     (delete-block! repo false)))
> ```

Thx. I just saw this comment. As I understand it, a journal should also be a page, and if so, then I think we can do without this `(or page (:block/journal? block))` condition? Am I right?

Reply:
Yes, that can be omitted.